<?xml version="1.0" encoding="UTF-8"?>
<interface>
  <!-- interface-requires gtk+ 3.10 -->
  <template class="GsUpgradeBanner" parent="GtkWidget">
    <child>
      <object class="GtkListBox">
        <property name="hexpand">True</property>
        <property name="vexpand">True</property>
        <property name="valign">center</property>
        <property name="margin-top">12</property>
        <style>
          <class name="content"/>
        </style>
        <child>
          <object class="GtkListBoxRow" id="box_upgrades_row">
            <property name="hexpand">True</property>
            <property name="vexpand">True</property>
            <property name="valign">center</property>
            <property name="activatable">False</property>
            <property name="selectable">False</property>
            <style>
              <class name="upgrade-banner-background"/>
            </style>
            <child>
              <object class="GtkBox">
                <property name="orientation">vertical</property>
                <property name="hexpand">True</property>
                <property name="vexpand">True</property>
                <property name="valign">center</property>
                <child>
                  <object class="GtkLabel" id="label_upgrades_title">
                    <property name="margin-top">32</property>
                    <property name="margin_bottom">6</property>
                    <!-- Just a placeholder; actual label text is set in code -->
                    <property name="label">GNOME 3.20 Now Available</property>
                    <attributes>
                      <attribute name="scale" value="2.0"/>
                      <attribute name="weight" value="ultrabold"/>
                    </attributes>
                  </object>
                </child>
                <child>
                  <object class="GtkLabel" id="label_upgrades_summary">
                    <property name="label" translatable="yes">A major upgrade, with new features and added polish.</property>
                    <property name="wrap">True</property>
                    <property name="width-chars">40</property>
                    <property name="max-width-chars">48</property>
                    <property name="margin-bottom">32</property>
                    <property name="halign">center</property>
                    <property name="justify">center</property>
                  </object>
                </child>
              </object>
            </child>
          </object>
        </child>
        <child>
          <object class="GtkListBoxRow">
            <property name="hexpand">True</property>
            <property name="vexpand">True</property>
            <property name="valign">center</property>
            <property name="activatable">False</property>
            <property name="selectable">False</property>
            <child>
              <object class="GtkBox">
                <property name="orientation">vertical</property>
                <property name="hexpand">True</property>
                <property name="vexpand">True</property>
                <style>
                  <class name="upgrade-banner-bottom-box"/>
                </style>

                <child>
                  <object class="GtkBox" id="box_upgrades_download">
                    <property name="orientation">vertical</property>
                    <property name="halign">center</property>
                    <property name="valign">center</property>
                    <property name="vexpand">True</property>
                    <property name="spacing">12</property>
                    <property name="margin-top">18</property>
                    <property name="margin-bottom">18</property>
                    <property name="margin-start">18</property>
                    <property name="margin-end">18</property>
                    <style>
                      <class name="upgrade-buttons"/>
                    </style>

                    <child>
                      <object class="GtkButton" id="button_upgrades_download">
                        <property name="label" translatable="yes">_Download</property>
                        <property name="width_request">150</property>
                        <property name="can_focus">True</property>
                        <property name="use_underline">True</property>
                        <property name="halign">center</property>
                        <property name="valign">center</property>
                        <property name="vexpand">True</property>
                        <style>
                          <class name="circular"/>
                          <class name="suggested-action"/>
                        </style>
                      </object>
                    </child>
                    <child>
                      <object class="GtkLabel" id="label_download_info">
                        <property name="label"></property> <!-- set in the code -->
                        <property name="halign">center</property>
                        <property name="valign">center</property>
                        <property name="vexpand">True</property>
                        <property name="justify">center</property>
                        <attributes>
                          <attribute name="scale" value="0.8"/>
                        </attributes>
                      </object>
                    </child>
                  </object>
                </child>
                <child>
                  <object class="GtkBox" id="box_upgrades_downloading">
                    <property name="orientation">horizontal</property>
                    <property name="halign">fill</property>
                    <property name="hexpand">True</property>
                    <property name="valign">center</property>
                    <property name="vexpand">True</property>
                    <property name="spacing">12</property>
                    <property name="margin-top">18</property>
                    <property name="margin-bottom">18</property>
                    <property name="margin-start">18</property>
                    <property name="margin-end">18</property>
                    <style>
                      <class name="upgrade-buttons"/>
                    </style>
                    <child>
                      <object class="GtkLabel" id="label_upgrades_downloading_spacer">
                        <property name="label"></property> <!-- space-taker -->
                        <property name="halign">start</property>
                      </object>
                    </child>
                    <child>
                      <object class="GtkBox" id="hbox_upgrades_downloading">
                        <property name="orientation">vertical</property>
                        <property name="halign">fill</property>
                        <property name="hexpand">True</property>
                        <property name="valign">center</property>
                        <property name="spacing">8</property>
                        <style>
                          <class name="upgrade-buttons"/>
                        </style>
                        <child>
                          <object class="GtkLabel" id="label_upgrades_downloading">
                            <property name="label"></property> <!-- set in the code -->
                            <property name="halign">center</property>
                            <property name="justify">center</property>
                            <attributes>
                              <attribute name="scale" value="0.8"/>
                              <attribute name="font-features" value="tnum=1"/>
                            </attributes>
                          </object>
                        </child>
                        <child>
                          <object class="GtkProgressBar" id="progressbar">
                            <property name="halign">center</property>
                            <property name="valign">center</property>
                            <property name="fraction">0.3</property>
                            <property name="width-request">340</property>
                            <style>
                              <class name="upgrade-progressbar"/>
                            </style>
                          </object>
                        </child>
                      </object>
                    </child>

                    <child>
                      <object class="GtkButton" id="button_upgrades_cancel">
                        <property name="can_focus">True</property>
                        <property name="use_underline">True</property>
                        <property name="halign">end</property>
                        <property name="hexpand">False</property>
                        <property name="valign">center</property>
                        <property name="margin-start">2</property>
                        <property name="margin-end">2</property>
                        <style>
                          <class name="circular"/>
                        </style>
                        <child>
                          <object class="GtkImage" id="button_upgrades_cancel_image">
                            <property name="icon-name">window-close-symbolic</property>
                            <property name="icon-size">normal</property>
                          </object>
                        </child>
                      </object>
                    </child>
                  </object>
                </child>
                <child>
                  <object class="GtkBox" id="box_upgrades_install">
                    <property name="orientation">vertical</property>
                    <property name="halign">center</property>
                    <property name="valign">center</property>
                    <property name="vexpand">True</property>
                    <property name="spacing">12</property>
                    <property name="margin-top">18</property>
                    <property name="margin-bottom">18</property>
                    <property name="margin-start">18</property>
                    <property name="margin-end">18</property>
                    <style>
                      <class name="upgrade-buttons"/>
                    </style>
                    <child>
                      <object class="GtkButton" id="button_upgrades_install">
                        <property name="label" translatable="yes">_Restart &amp; Upgrade</property>
                        <property name="name">button_upgrades_install</property>
                        <property name="can_focus">True</property>
                        <property name="use_underline">True</property>
                        <property name="halign">center</property>
                        <property name="valign">center</property>
                        <style>
                          <class name="circular"/>
                          <class name="suggested-action"/>
                        </style>
                      </object>
                    </child>
                    <child>
                      <object class="GtkLabel" id="label_upgrade_warning">
                        <property name="label" translatable="yes">Remember to back up data and files before upgrading.</property>
                        <property name="halign">center</property>
                        <property name="justify">center</property>
                        <attributes>
                          <attribute name="scale" value="0.8"/>
                        </attributes>
                      </object>
                    </child>
                  </object>
                </child>
              </object>
            </child>
          </object>
        </child>
      </object>
    </child>
  </template>

  <object class="GtkSizeGroup">
    <property name="mode">vertical</property>
    <widgets>
      <widget name="box_upgrades_download"/>
      <widget name="box_upgrades_downloading"/>
      <widget name="box_upgrades_install"/>
    </widgets>
  </object>
</interface>
